### DISCUSSION TITLE
Add support for LilyGO T-Echo

### DISCUSSION THREAD

**Discussion by malteish** [markqvist/Reticulum#642] [Ideas]
The [LilyGo T-Echo](https://lilygo.cc/products/t-echo-lilygo) is very popular right now. Adding support for it might make RNode more accessible to a larger user base.

![image](https://github.com/user-attachments/assets/f69d6a89-33fa-4dbf-823c-6c9e4f0d3c03)

**Comment by cobraPA** [comment]
The new(ish) Rnode CE firmware has T Echo support, don't know if this link is too deep for you: https://github.com/liberatedsystems/RNode_Firmware_CE/blob/master/Boards.h.  I'm not up to date if the web flasher supports it as well, but I believe the firmware is in a working state.

**Comment by antonionardella** [comment]
Hi,

yesterday I flashed my T-Echo using the online flasher, and I have to say—it’s amazing!

However, I’ve encountered a few issues that I’d like to discuss:

- The display isn’t fully utilized, making it difficult to read what’s happening
- The radio stops working when disconnected from USB, meaning it doesn’t seem to be using the battery properly
- The grey status carousel (showing version and hardware details) continuously consumes power, which isn’t ideal for an e-ink display

<img src="https://github.com/user-attachments/assets/64fa2c24-e754-441c-8bff-d6d95f553e13" alt="drawing" width="200"/>

I’m very new to this and not sure if I can contribute code-wise, but I’d love to explore potential fixes and test things out. Could anyone point me in the right direction on where to look in the code and how I might improve these aspects of the T-Echo?

Thanks in advance!

**Reply by markqvist** [reply]
What version did you use, and was it the CE firmware or the "official" one?

(judging from the backlight intensity, it looks like it's an older or CE version)

**Reply by antonionardella** [reply]
Hi and thank you for getting back to me. According to the [flasher page](https://liamcottle.github.io/rnode-flasher/) it's version 1.81
![image](https://github.com/user-attachments/assets/2505c3f0-531e-4551-90c6-5022a789b2c3)

I also tried to use `rnodeconf` and it also pulled the 1.81 version
![image](https://github.com/user-attachments/assets/a32fc6cf-6191-46ea-8c1e-f06e192f00a0)

I reviewed the instructions from both repositories, and they mention using `rnodeconf --autoinstall` to set up the node. However, I’m unsure how to choose between the Community Edition (CE) firmware and the official firmware.

Could you clarify the difference and how to select the appropriate version? Apologies for any confusion on my part!

Thanks! 🙏

**Reply by markqvist** [reply]
No worries, I thought you might have used the CE version, which I think has the problem of not working on battery power. In version 1.81 it works as it should, and the device runs without USB power attached. So if it is not powering up on battery power on version 1.81, I'm a bit confused what is going on.

**Reply by antonionardella** [reply]
I was running Meshtastic successfully on this device before flashing the rnode firmware and can confirm that the hardware is working

<img src="https://github.com/user-attachments/assets/7e5b84f5-5d59-4019-8165-f428ced5fb80" alt="drawing" width="200"/>
<img src="https://github.com/user-attachments/assets/662fd465-89f5-4f99-9f9a-e38452845ac8" alt="drawing" width="200"/>
<img src="https://github.com/user-attachments/assets/ba361366-8dad-4f73-8795-81f9afc47814" alt="drawing" width="200"/>

----

UPDATE:

Alright, I reflashed meshtastic and there I also had the issue with the battery, so I opened the device and the battery was disconnected. Once the battery has been reconnected the T-Echo started working without USB.

Now I reflashed the rnode firmware and it also works via battery.

Sorry for the confusion.

About the other two things, like the carousel and display size, are there any possible things to look at?

Thank you

**Comment by PetzGordo** [comment]
Somehow I don't understand which button does what. Does the capacitive button have any function? 
And is there a way to switch off the backlight? It is permanently lit on my devices (both have FW v1.81).
Thank you!

**Reply by markqvist** [reply]
The backlight and screen blanking timeout can be configured with `rnodeconf`.

The capacitive button is unused currently. One of the physical buttons is a hardwired reset button, it performs a reset. The other is used as the user button for bluetooth control, sleep mode and other things.
